src/parser/raindocument.rs

```diff
@@ -0,0 +1,1746 @@
+use lsp_types::Url;
+use std::sync::{Arc, RwLock};
+use serde::{Serialize, Deserialize};
+use async_recursion::async_recursion;
+use topo_sort::{SortResults, TopoSort};
+use std::collections::{HashMap, VecDeque};
+use futures::{executor::block_on, future::join_all};
+use rain_meta::{
+    Store, NPE2Deployer, KnownMagic, RainMetaDocumentV1Item, search_deployer, search,
+    types::{
+        dotrain::v1::DotrainMeta, authoring::v1::AuthoringMeta,
+        interpreter_caller::v1::InterpreterCallerMeta,
+    },
+};
+use super::{
+    rainlangdocument::RainlangDocument,
+    super::{
+        error::Error,
+        types::{ast::*, patterns::*},
+    },
+    exclusive_parse, inclusive_parse, fill_in, is_consumable, tracked_trim, line_number, to_u256,
+};
+
+#[cfg(any(feature = "js-api", target_family = "wasm"))]
+use tsify::Tsify;
+#[cfg(any(feature = "js-api", target_family = "wasm"))]
+use wasm_bindgen::prelude::wasm_bindgen;
+
+/// Reserved constant values keys in RainDocuments
+pub const RAIN_DOCUMENT_CONSTANTS: [(&str, &str); 9] = [
```

Comment:
do we still need reserved constants in .rain files, or is this covered in rainlang now by @thedavidmeister's subparser work?

Reply:
these dotrain's, which are different from those subparser constants